hera_mc/cm_hookup.py

```diff
@@ -415,78 +89,90 @@ def get_hookup(self, hpn_list, rev='ACTIVE', port_query='all', at_date='now', ex
         self.at_date = at_date
         self.hookup_type = hookup_type
 
-        # Take appropriate action if hpn_list is a string
-        if isinstance(hpn_list, six.string_types):
-            if hpn_list.lower().startswith('cache'):
-                print("Force read of cache file - not guaranteed fresh.")
-                self.read_hookup_cache_from_file()
-                return self.cached_hookup_dict
-            else:
-                hpn_list = cm_utils.listify(hpn_list)
-
-        # Check if force_db either requested or needed
-        if force_db or not self._requested_list_OK_for_cache(hpn_list):
-            return self.get_hookup_from_db(hpn_list=hpn_list, rev=rev, port_query=port_query, at_date=at_date,
-                                           exact_match=exact_match, hookup_type=hookup_type)
+        if isinstance(hpn, six.string_types) and hpn.lower() == 'cache':
+            self.read_hookup_cache_from_file()
+            return self.cached_hookup_dict
 
-        # Check/get the appropriate hookup dict
-        # (a) in memory, (b) re-read cache file, or (c) generate new
-        self._hookup_cache_to_use(force_new_cache=force_new_cache)
+        hpn = cm_utils.listify(hpn)
+        if use_cache and self.requested_list_OK_for_cache(hpn):
+            self.read_hookup_cache_from_file()
+            return self._cull_dict(hpn, self.cached_hookup_dict, exact_match)
 
-        # Now build up the returned hookup_dict
-        hookup_dict = {}
-        hpn_list = [x.lower() for x in hpn_list]
-        for k in self.cached_hookup_dict:
-            hpn, rev = cm_utils.split_part_key(k)
-            use_this_one = False
-            if exact_match:
-                if hpn.lower() in hpn_list:
-                    use_this_one = True
-            else:
-                for p in hpn_list:
-                    if hpn[:len(p)].lower() == p:
-                        use_this_one = True
-                        break
-            if use_this_one:
-                hookup_dict[k] = copy.copy(self.cached_hookup_dict[k])
-        return hookup_dict
+        return self.get_hookup_from_db(hpn=hpn, pol=pol, at_date=at_date,
+                                       exact_match=exact_match, hookup_type=hookup_type)
```

Comment:
This line calls a method that is further down the file. Methods that are called by other methods should always appear earlier in the file, before they are called.

Reply:
Moved the method.